packages/core/src/queries/resource.ts

```diff
@@ -26,6 +26,35 @@ export const createResourceQueries = (pool: CommonQueryMethods) => {
       where ${fields.indicator}=${indicator}
     `);
 
+  const findDefaultResource = async () =>
+    pool.maybeOne<Resource>(sql`
+      select ${sql.join(Object.values(fields), sql`, `)}
+      from ${table}
+      where ${fields.isDefault}=true
+    `);
+
+  const setDefaultResource = async (id: string) => {
+    return pool.transaction(async (connection) => {
+      await connection.query(sql`
+        update ${table}
+          set ${fields.isDefault}=false
+          where ${fields.id}!=${id};
+      `);
```

Comment:
Not sure which one is better. Will querying out the current default resource be more efficient?

Reply:
i think they are similar since `isDefault` is not indexed. but yours is better.